turbomoleio/core/periodic.py

```diff
@@ -0,0 +1,240 @@
+            periodic_str = dg.sdg("$periodic", strict=True)
+            if periodic_str is None:
+                raise ValueError('The $periodic data group should be set for periodic systems.')
+            if cell_str is None and lattice_str is None:
+                raise ValueError('The $cell or $lattice data group should be set for periodic systems.')
+            if cell_str is not None and lattice_str is not None:
+                raise ValueError('Only one of $cell and $lattice data group should be set for periodic systems.')
+            mol, fi = get_mol_and_indices_frozen(coordinates_str, cell_string=cell_str,
+                                                 lattice_string=lattice_str,
+                                                 periodic_string=periodic_str,
+                                                 periodic_extension=periodic_extension)
+            periodicity = int(periodic_str.strip())
+
+            int_def_str = dg.sdg("$intdef", strict=True)
+            if int_def_str is not None:
+                raise ValueError('Internal definitions for periodic systems is not supported.')
+
+            user_def_bonds_str = dg.sdg("$user-defined bonds", strict=True)
+            if user_def_bonds_str is not None:
+                raise ValueError('User-defined bonds for periodic systems is not supported.')
+
+            return cls(mol, frozen_indices=fi, periodicity=periodicity)
+
+        else:
+            return cls(Structure.from_str(string, fmt))
+
+    @classmethod
+    def from_file(cls, filepath, fmt=None, periodic_extension=5.0):
+        """
+        Creates an instance from a file. Could be a coord file or any other
+        format supported by pymatgen Structure.
+
+        Args:
+            filepath (str): path to the file.
+            fmt (str): the format of the data. could be "coord" for Turbomole
+                coord file or any format supported in pymatgen Structure.
+                If None the code would try to infer it from the file name.
+
+        Returns:
+            An instance of PeriodicSystem.
+        """
+        fname = os.path.basename(filepath)
+        if fmt == "coord" or (fmt is None and fnmatch(fname, "coord")):
+            with open(filepath) as f:
+                return cls.from_string(f.read(), fmt="coord", periodic_extension=periodic_extension)
+        else:
+            return cls(structure=Structure.from_file(filepath))
+
+    def to_coord_string(self):
+        """
+        Creates the string of a coord file for this system.
+
+        Returns:
+            str: the string representing the system.
+        """
+        lines = ["$coord"]
+        lines.extend(self.coord_lines())
+
+        lines.append("$periodic {}".format(self.periodicity))
+        lattice = self.structure.lattice
+        lmat = lattice.matrix
+        if (
+                not np.allclose(lmat[0][1:], 0.0) or
+                not np.isclose(lmat[1][2], 0.0)
+        ):
```

Comment:
if the $lattice datagroup is supported in reading it should likely be supported in writing as well. It seems that a structure that does not respect the $cell convention will be blocked here before writing it down, but the incosistency should be addressed at some point

Reply:
How would you do it ? Having by default the lattice that is written ? With an argument in the to_coord_string method which says whether to use $cell or $lattice ? We can probably also postpone that in a future PR ?

Reply:
In general I would just use the most general case (i.e. $lattice in this case), which should also be easier to implment. Do you think there is a use case that requires to keep both options? 

In any case we can postpone it to another PR. Maybe open an issue to keep track of this.

Reply:
Ok. Added issue #16